fix(commands): fix `remove_cluster` command and add unit tests for BC and area removal

laurent-laporte-pro opened the pull request:
Description:

This Pull Request addresses a bug in the `remove_cluster` command and enhances our test coverage for the management of binding constraints and the removal of areas in studies of versions 7.2 and 8.7.

Changes include:

1. **Bug Fix in `remove_cluster` command**: The command now correctly handles the removal of binding constraints matrices in version 8.7 when a thermal cluster is deleted.

2. **Unit Tests for Binding Constraints**: Added new unit tests to verify the correct handling of binding constraints.

3. **Unit Tests for Area Removal**: Introduced new unit tests for the removal of areas in studies.

laurent-laporte-pro committed:
fix(commands): added support for removing BC matrices in v8.7 when deleting a thermal cluster

laurent-laporte-pro committed:
test(commands): improve unit tests on BC management

laurent-laporte-pro committed:
test(commands): improve unit tests for `remove_area` command

MartinBelthle commented on tests/variantstudy/model/command/test_manage_binding_constraints.py:
```diff
@@ -83,10 +64,18 @@ def test_manage_binding_constraint(
+        ]
+    for matrix_link in matrix_links:
+        link_path = study_path / f"input/bindingconstraints/{matrix_link}"
+        assert link_path.exists(), f"Missing matrix link: {matrix_link!r}"
```
Why do you add this assertion ? This comment also applies to the same assertions below.

From my understanding, this check is useless and can even hide some bugs. So i'm probably missing something.

laurent-laporte-pro replied:
The previous `apply` generates the snapshot. I want to make sure the default matrices are generated with the right name(s).

MartinBelthle commented on tests/variantstudy/model/command/test_remove_area.py:
```diff
@@ -18,13 +17,8 @@
-        command_context: CommandContext,
-        version: int,
-    ):
+    @pytest.mark.parametrize("empty_study", ["empty_study_810.zip", "empty_study_840.zip"], indirect=True)
```
This new way to implement the fixture is nice :)

laurent-laporte-pro replied:
💪😀

MartinBelthle commented on tests/variantstudy/model/command/test_remove_cluster.py:
```diff
@@ -13,6 +15,7 @@
 
 
 class TestRemoveCluster:
+    @pytest.mark.parametrize("empty_study", ["empty_study_720.zip", "empty_study_870.zip"], indirect=True)
```
It could be good to add an integration test to ensure the endpoint deletes clusters properly.

laurent-laporte-pro replied:
Done

MartinBelthle commented on antarest/study/storage/variantstudy/model/command/remove_cluster.py:
```diff
@@ -134,26 +134,50 @@ def _create_diff(self, other: "ICommand") -> t.List["ICommand"]:
+                    bc_to_remove[bc_index] = binding_constraints.pop(bc_index)
+                    break
+
+        matrix_suffixes = ["_lt", "_gt", "_eq"] if study_data.config.version >= 870 else [""]
```
Indeed, i forgot to change this. Good job :)

laurent-laporte-pro replied:
Missing: `study_data.config.bindings.remove(bc)`!

laurent-laporte-pro replied:
The update of the configuration is already performed in `remove_area_cluster_from_binding_constraints`, when the configuration is _applied_.

laurent-laporte-pro committed:
feat(commands): binding constraint commands can accept matrix names in camelCase

laurent-laporte-pro committed:
test(api-thermals): add case with a BC referencing a thermal

laurent-laporte-pro commented:
I have added aliases to the parameters of the Binding Constraint (BC) commands:

- `lessTermMatrix`,
- `greaterTermMatrix`
- `equalTermMatrix`.

This way, we can have either snake_case or camelCase.

I also added an integration test in the Thermals to test the deletion of a cluster that is referenced by a binding constraint.
